### ISSUE TITLE
If one .populate() call fails, all consequent .populate() calls fail

### ISSUE THREAD

**Issue by Alaneor** [balderdashy/waterline#806]
I have a model with two one-to-many associations in a MySQL database. One of the associated models is stored in a table where the primary (and non-auto-incrementing) key is also a foreign key to the parent model.

I did not expect that association to work properly, so that's fine. However, when I attempt to populate both of them from the parent model, if the first `.populate()` call goes to the broken association, the next `.populate()` call to a standard (and working) association also fails (i.e. I do not see the data, only an empty array).

Here's a simple example.

```js
User    // Parent model. Normal db definition, all is well
  .findOne(123)
  .populate('payments') // This is the assoc to a non-well designed table. I expect it to fail
  .populate('goods')  // Assoc to a well-designed table. I expect this to work
  .exec(function (err, users){
    // Nothing here
    if (err) throw err
    console.log(users.payments) // Empty array - expected, table definition sucks
    console.log(users.goods)  // Empty array - UNEXPECTED
  })
```

However, if I change the order of the `.populate()` calls then at least the first association is resolved correctly (i.e. I see the data in `users.goods`).

**Likely cause:**
I have not looked at the source code, but it feels as if there was a call to `async.series` and an error in the first item in the stack will fail all consecutive stacks (or, rather, skip them completely).

Let me know if I can provide any additional troubleshooting!

**Comment by devinivy** [comment]
Thanks, @Alaneor.  I realize this issue is somewhat old, but I'd like to look into this.  Is there any possibility you could write a failing test in [here](https://github.com/balderdashy/waterline-adapter-tests/blob/master/interfaces/associations/manyToMany/findOne.populate.js) in waterline-adapter-test?

**Comment by CWyrtzen** [comment]
any update on this?

**Comment by devinivy** [comment]
In this case, it seems an error should be passed and as much of the result set as possible should be returned.  That said, this may be impossible to achieve with native joins, as in MySQL where an entire query could be destroyed by one bad association.

**Comment by sailsbot** [comment]
Thanks for posting, @Alaneor.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
